config/configmapprovider/provider.go

```diff
@@ -40,15 +45,27 @@ type Provider interface {
 	// contains the value to be injected in the configuration and the corresponding watcher that
 	// will be used to monitor for updates of the retrieved value.
 	//
-	// watcher callback is called when the config changes. watcher may be called from
+	// `location` must follow the "<scheme>:<opaque_data>" format. This format is compatible with the URI definition
+	// (https://datatracker.ietf.org/doc/html/rfc3986). If the "<scheme>" is one of the well known URI scheme names
+	// (see https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml) then the implementation should follow
+	// the standard format to avoid confusions.
+	//
+	// `watcher` callback is called when the config changes. watcher may be called from
 	// a different go routine. After watcher is called Retrieved.Get should be called
 	// to get the new config. See description of Retrieved for more details.
 	// watcher may be nil, which indicates that the caller is not interested in
 	// knowing about the changes.
 	//
 	// If ctx is cancelled should return immediately with an error.
 	// Should never be called concurrently with itself or with Shutdown.
-	Retrieve(ctx context.Context, watcher WatcherFunc) (Retrieved, error)
+	Retrieve(ctx context.Context, location string, watcher WatcherFunc) (Retrieved, error)
+
+	// Scheme returns the implemented scheme name (see https://datatracker.ietf.org/doc/html/rfc3986#section-3.1)
+	// by this Provider.
+	//
+	// Implementation must make sure to not conflict with well known scheme names, or follow the standard,
+	// see https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
+	Scheme() string
```

Comment:
This doesn't seem to be used anywhere and it's not clear to me where it would be useful.  Are there plans for it or can it be removed?

Reply:
Good call, can be added later. I will remove it for the moment.